### PR TITLE
Abductor chem dispenser

### PR DESCRIPTION
**Pull request by XDTM**
## About The Pull Request

Adds an abductor chem dispenser, buyable for 2 points from the abductor rewards console.
Buying it spawns a beacon, which can be activated in hand to drop it and spawn the dispenser after 3 seconds if the tile is free of obstructions.
The dispenser contains all the reagents of a fully upgraded, emagged dispenser plus plasma and uranium.

## Why It's Good For The Game

More tools for abductors means more possible abductor shenanigans, and personally i'm all for making the no-murderbone antag as fun as possible.

## Changelog
:cl: XDTM
add: Abductors can now buy a special chem dispenser for 2 points. The machine is delivered as a beacon which can be used when standing on any free tile to spawn it.
/:cl:

### CODE REVIEW

**Comment by HomocidlePancakes** [comment]
Don't forget complimentary beakers.

**Comment by Fikou** [comment]
REEE i wanted to put a thing like that in my advanced abductor ship

**Comment by Dawson1917** [comment]
This makes it very easy for abductors to make romerol zombies. Might not be too bad but knowing how often abductors try to griff you may want to remove the emagged chems from the abductor dispenser if it happens every time they spawn

**Comment by XDTM** [comment]
Good point, there should be a little effort involved. I'll remove the carpotoxin.

**Comment by Fikou** [comment]
uhh isnt removing carpotoxin from emagged chem dispensers a balance update

**Comment by XDTM** [comment]
only from the new one i made for abductors, don't worry

**Comment by FantasticFwoosh** [comment]
Carpotoxin is not difficult to procure, especially if there are pre-existing sources which are pretty common on the station such as carp, trace amounts in chef's food & koibeans. It stands to reason that this feels a bit strange when we had chemistry dispenser upgrade disks a while back (last year?), to make it abductor exclusive too.

Do abductors really need self-made chem grenades, choral hydrate or any form of chemistry to further their objectives other than grief? I thought light amount of looting medkits was part and parcel to their infiltration.

**Comment by HomocidlePancakes** [comment]
>Carpotoxin is not difficult to procure, especially if there are pre-existing sources which are pretty common on the station such as carp, trace amounts in chef's food & koibeans. It stands to reason that this feels a bit strange when we had chemistry dispenser upgrade disks a while back (last year?), to make it abductor exclusive too.

Ok. Look buddy. If you leave a loaded gun in the same room as a 10 year-old child (The average psychological age of our playerbase), he will shoot himself, BUT, if you leave an UNLOADED GUN in the same room as him and put the ammo on the highest shelf in the kitchen instead he would rather HAVE FUN with the unloaded gun instead of going throught the pain of STACKING FLIMSY CHAIRS on top one another.

>Do abductors really need self-made chem grenades

Yas. Also look two points down.

>choral hydrate 

It's shit and been like that for fuck knows how long.

> or any form of chemistry to further their objectives other than grief?

Their entire objective is to grief. 

>I thought light amount of looting medkits was part and parcel to their infiltration.

Thats the bare minimum. Whats the fun in doing bare minimum?

Ok. Look. The whole point of abductors is to grief people. Mindcontrol, brainwashing, glands that give silly powers - what purpose do these things serve other then grief? Abductor is the only antag in the game that completly relies on interacting with other people to have fun. Traitor will just kill you with no counterplay and leave you out of the game for 2 hours (fun, right?). Abductor is going to cut off all of your limbs, make you unable to say anything besides "NUGGET!" and drop you off infront of AI just to make it freak out. And thats why I like them so much.

**Comment by ishitbyabullet** [comment]
Honestly, they don't need a chem dispenser.

I thought that abductors were one of the few non-lethal antags we have though. I remember someone being punished for abducting someone then setting the dropoff point for all abductees to space so they would die while being unable to escape.

If they truly are supposed to be non-lethal then them using a chem grenade would probably require the same reasoning as using their powerful cloning damage gun.

**Comment by HomocidlePancakes** [comment]
Not all grenades have to be explosive.

**Comment by datorangebottle** [comment]
> Honestly, they don't need a chem dispenser.

Abductors need the dispenser for healing, mostly. Also, abductors using the grenades to lethal people will obviously be jobbeaned or whatnot. Lethals are pretty strictly not allowed except in emergencies, as I recall.

**Comment by OverlyGeneric** [comment]
@XDTM There is a flashing stray pixel in the top-left corner of the abductor chem dispenser's sprite.
image

**Comment by optimumtact** [comment]
This is absurd power creep, you'll have to come up with more than "it's cool" as a justification for a 2 point reward.

**Comment by CRITAWAKETS** [comment]
What purpose does having an abductor chem dispenser serve other than buffing murderboning? Especially since you are adding the toxins and plasma to it, which only serve to make things such as Lexorin and TOXINS. This is just powercreep for murderboning.

**Comment by XDTM** [comment]
I mostly assumed that power creeping abductors would be OK since their driving objective is just messing with people, and adding more tools gives them more potential creativity.
It can increase their survivability after 2 abductions, but most abductors steal medical supplies anyway, and crafter ones can easily build one if they want to.

@CRITAWAKETS abductors are forbidden from murder boning, which is why I was so carefree about the buff. I added plasma and uranium as the usual abductor-tech-is-more-advanced reasoning.

**Comment by XDTM** [comment]
The price can also be adjusted so buying one can mean not getting other useful tools like the mind device or backup batons

**Comment by optimumtact** [comment]
I'm open to that I suppose, but it does feel a little uncreative to just say, oh they can use chems to grief now so wacky

**Comment by Detective-Google** [comment]
Abductor chems isn't that awful but it's just.... Why? They already have a super stun baton, why do they need more? 

Also yeah, having to scavenge meds, or get risky for gimmicks is part of the fun

**Comment by XDTM** [comment]
A chem dispenser is not the most creative of tools by itself, that is true.

My ideal for abductors is for them to be able to gradually expand their starting toolset by spending credits, either by buying more gizmos or by getting more machinery.
Technically, they could get similar stuff with station supplies; but i believe that it's better to have them spend more time abducting to gain credits (interacting with the crew while also being vulnerable) than doing tele-thievery (which is really hard to counter). Plus i find the whole abductor™ tech aestethic appealing, and i'd like to add some mechanic to rarely allow the station to get their hands on that stuff.

**Comment by ishitbyabullet** [comment]
I like the idea of more abductor-themed things existing due to how much abductor stuff there already is.

**Comment by Shadowflame909** [comment]
I always wanted more buyable abductor items.

**Comment by habader** [comment]
put it at like 4-5 points

**Comment by habader** [comment]
fucking finally

File: code/modules/antagonists/abductor/equipment/abduction_gear.dm [hunk 1]
```diff
@@ -639,6 +639,42 @@ Congratulations! You are now trained for invasive xenobiology research!"}
...
+		var/new_machine = new spawned_machine(loc)
+		visible_message("<span class='notice'>[new_machine] warps on top of the beacon!")
+		qdel(src)
+	else
+		playsound(src, 'sound/machines/buzz-two.ogg', 50, 0)
+
+/obj/item/abductor_machine_beacon/chem_dispenser
+	name = "beacon - Reagent Synthetizer"
+	desc = "A beacon designed to instantly tele-construct abductor machinery."
+	w_class = WEIGHT_CLASS_TINY
```
Comment (SpaceManiac) [inline code/modules/antagonists/abductor/equipment/abduction_gear.dm]: `desc` and `w_class` are duplicate

File: code/modules/antagonists/abductor/equipment/abduction_gear.dm [hunk 2]
```diff
@@ -639,6 +639,42 @@ Congratulations! You are now trained for invasive xenobiology research!"}
...
+	if(viable)
+		playsound(src, 'sound/effects/phasein.ogg', 50, 1)
+		var/new_machine = new spawned_machine(loc)
+		visible_message("<span class='notice'>[new_machine] warps on top of the beacon!")
+		qdel(src)
+	else
+		playsound(src, 'sound/machines/buzz-two.ogg', 50, 0)
+
+/obj/item/abductor_machine_beacon/chem_dispenser
+	name = "beacon - Reagent Synthetizer"
```
Comment (SpaceManiac) [inline code/modules/antagonists/abductor/equipment/abduction_gear.dm]: Do you mean "Synthesizer"? (elsewhere also)

**Review by SpaceManiac — CHANGES_REQUESTED** [review]

**Review by SpaceManiac — DISMISSED** [review]

File: code/modules/antagonists/abductor/equipment/abduction_gear.dm [hunk 3]
```diff
@@ -639,6 +639,40 @@ Congratulations! You are now trained for invasive xenobiology research!"}
...
+/obj/item/abductor_machine_beacon/proc/try_spawn_machine()
+	var/viable = FALSE
+	if(isfloorturf(loc))
+		var/turf/T = loc
+		viable = TRUE
+		for(var/obj/thing in T.contents)
+			if(thing.density || istype(thing, /obj/machinery) || istype(thing, /obj/structure))
+				viable = FALSE
+	if(viable)
+		playsound(src, 'sound/effects/phasein.ogg', 50, 1)
```
Comment (ShizCalev) [inline code/modules/antagonists/abductor/equipment/abduction_gear.dm]: TRUE

File: code/modules/antagonists/abductor/equipment/abduction_gear.dm [hunk 4]
```diff
@@ -639,6 +639,40 @@ Congratulations! You are now trained for invasive xenobiology research!"}
...
+		var/new_machine = new spawned_machine(loc)
+		visible_message("<span class='notice'>[new_machine] warps on top of the beacon!")
+		qdel(src)
+	else
+		playsound(src, 'sound/machines/buzz-two.ogg', 50, 0)
```
Comment (ShizCalev) [inline code/modules/antagonists/abductor/equipment/abduction_gear.dm]: remove 0

File: code/modules/antagonists/abductor/equipment/abduction_gear.dm [hunk 5]
```diff
@@ -639,6 +639,40 @@ Congratulations! You are now trained for invasive xenobiology research!"}
...
+	playsound(src, 'sound/machines/terminal_alert.ogg', 50, 0)
+	addtimer(CALLBACK(src, .proc/try_spawn_machine), 30)
+
+/obj/item/abductor_machine_beacon/proc/try_spawn_machine()
+	var/viable = FALSE
+	if(isfloorturf(loc))
+		var/turf/T = loc
+		viable = TRUE
+		for(var/obj/thing in T.contents)
+			if(thing.density || istype(thing, /obj/machinery) || istype(thing, /obj/structure))
```
Comment (ShizCalev) [inline code/modules/antagonists/abductor/equipment/abduction_gear.dm]: ismachinery & isstructure

File: code/modules/antagonists/abductor/equipment/abduction_gear.dm [hunk 6]
```diff
@@ -639,6 +639,40 @@ Congratulations! You are now trained for invasive xenobiology research!"}
...
+	icon = 'icons/obj/abductor.dmi'
+	icon_state = "beacon"
+	w_class = WEIGHT_CLASS_TINY
+	var/obj/machinery/spawned_machine
+
+/obj/item/abductor_machine_beacon/attack_self(mob/user)
+	..()
+	user.visible_message("<span class='notice'>[user] places down [src] and activates it.", "<span class='notice'>You place down [src] and activate it.")
+	user.dropItemToGround(src)
+	playsound(src, 'sound/machines/terminal_alert.ogg', 50, 0)
```
Comment (ShizCalev) [inline code/modules/antagonists/abductor/equipment/abduction_gear.dm]: remove 0

File: code/modules/antagonists/abductor/equipment/abduction_gear.dm [hunk 7]
```diff
@@ -639,6 +639,40 @@ Congratulations! You are now trained for invasive xenobiology research!"}
...
+	name = "machine beacon"
+	desc = "A beacon designed to instantly tele-construct abductor machinery."
+	icon = 'icons/obj/abductor.dmi'
+	icon_state = "beacon"
+	w_class = WEIGHT_CLASS_TINY
+	var/obj/machinery/spawned_machine
+
+/obj/item/abductor_machine_beacon/attack_self(mob/user)
+	..()
+	user.visible_message("<span class='notice'>[user] places down [src] and activates it.", "<span class='notice'>You place down [src] and activate it.")
```
Comment (ShizCalev) [inline code/modules/antagonists/abductor/equipment/abduction_gear.dm]: missing closing spans

**Review by ShizCalev — CHANGES_REQUESTED** [review]

**Review by SpaceManiac — DISMISSED** [review]

**Comment by adamkad1** [comment]
does zombie surgery actually make romerol zombies? i though it makes normal zombies (moans and such but no claws)

**Comment by XDTM** [comment]
It's the real deal

File: code/modules/antagonists/abductor/machinery/console.dm [hunk 8]
```diff
@@ -45,6 +45,7 @@
 		dat += "<b>Transfer data in exchange for supplies:</b><br>"
 		dat += "<a href='?src=[REF(src)];dispense=baton'>Advanced Baton (2 Credits)</A><br>"
 		dat += "<a href='?src=[REF(src)];dispense=mind_device'>Mental Interface Device (2 Credits)</A><br>"
+		dat += "<a href='?src=[REF(src)];dispense=chem_dispenser'>Reagent Synthetizer (2 Credits)</A><br>"
```
Comment (qwertyquerty) [inline code/modules/antagonists/abductor/machinery/console.dm]: "Synthetizer"

Comment (SpaceManiac) [reply code/modules/antagonists/abductor/machinery/console.dm]: fix it then

**Pull request merged**